jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -109,12 +107,117 @@ class AdministrationControllerTest extends Specification {
             def e = thrown(ValidationException)
             [errorMessage] == e.defaultErrorMessages
         and: 'Group is not created'
-            groups.assertDoesNotExist(group)
+            groups.assertDoesNotExistWithName(group.name)
         where:
             groupName           | groupDescription    |errorMessage                                                    | caseName
             ''                  | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name is empty'
             '     '             | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name contains only spaces'
             alphanumeric(101)   | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name too long'
             alphanumeric(1,100) | alphanumeric(256)   |"Group description length must be between 0 and 255 characters" | 'Group description too long'
     }
+
+    def 'must not be able to edit group without admin permission'() {
+        given: 'User signed in and forum has some existing group'
+            def session = users.signInAsRegisteredUser(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'User attempts to edit an existing group'
+            def dto = new GroupAdministrationDto()
```

Comment:
Can you create a separate method to generate random DTOs? The test is about permissions while the majority of lines are about creating the DTO :)